applications/MeshingApplication/python_scripts/modelers/convert_linear_tets_to_quadratic_modeler.py

```diff
@@ -0,0 +1,107 @@
+    """
+    This class is a modeler in KratosMultiphysics that converts linear tetrahedral (tet) elements to quadratic tet elements within a specified model part.
+
+    Attributes:
+        model (KratosMultiphysics.Model): The Kratos model containing the model part to be modified.
+        settings (KratosMultiphysics.Parameters): Settings for the conversion process.
+
+    Methods:
+        __init__(model, settings): Constructor for the class.
+        SetupModelPart(): Performs the conversion from linear to quadratic tet elements.
+        __GetDefaultSettings(): Provides default settings for the modeler.
+        __str__(): String representation of the class.
+        Factory(model, settings): Factory method to create an instance of the class.
+    """
+
+    def __init__(self, model, settings):
+        """
+        Constructor for the ConvertLinearTetsToQuadraticModeler class.
+
+        Args:
+            model (KratosMultiphysics.Model): The Kratos model containing the model part to be modified.
+            settings (KratosMultiphysics.Parameters): Configuration settings for the modeler.
+
+        The constructor initializes the modeler with the provided model and settings, and validates and assigns default settings.
+        """
+        super().__init__(model, settings)
+        settings.ValidateAndAssignDefaults(self.__GetDefaultSettings())
+        self.model = model
+        self.settings = settings
+
+    def SetupModelPart(self):
+        """
+        Converts the specified model part's linear tet elements to quadratic tets.
+
+        Raises:
+            Exception: If the operation is attempted in a distributed (MPI) run or if the input elements are not linear tets (4 nodes).
+
+        This method checks the model part for compatibility, sets necessary variables, and calls the utility function for conversion. It also handles the replacement of elements and conditions based on provided settings.
+        """
+        super().SetupModelPart()
+
+        if KratosMultiphysics.IsDistributedRun():
+            raise Exception("ConvertLinearTetsToQuadraticModeler is not MPI-ready")
+
+        model_part = self.model[self.settings["model_part_name"].GetString()]
+
+        for elem in model_part.Elements:
+            if elem.GetGeometry().GetGeometryType() != KratosMultiphysics.GeometryData.KratosGeometryType.Kratos_Tetrahedra3D4:
+                raise Exception("Only linear tets are accepted as input (4 nodes)")
+
+        KratosMultiphysics.VariableUtils().SetNonHistoricalVariable(
+            KratosMultiphysics.SPLIT_ELEMENT,
+            True,
+            model_part.Elements)
+        KratosMultiphysics.VariableUtils().SetNonHistoricalVariable(
+            KratosMultiphysics.SPLIT_ELEMENT,
+            True,
+            model_part.Conditions)
+
+        Meshing.LinearToQuadraticTetrahedraMeshConverter(model_part.GetRootModelPart()).LocalConvertLinearToQuadraticTetrahedraMesh(False, False)
```

Comment:
In here we're silently taking the root model part from the user specified one. This means that we could be silently converting all the elements in the root, which might be more than the ones the user is expecting.

I agree with the idea of enforcing the use of the root model part in this modeler to avoid an extremely complicated implementation (at least until it is otherwise required). Having said this, my suggestion would be to throw an error if the user-provided model part is not the root to inform the user about the expected use of this.